bin/waagent2.0

```diff
@@ -3300,6 +3300,87 @@ class Certificates(object):
             filename = str(index) + ".prv"
         return self
 
+
+class RemoteAccess(object):
+    """
+    Parse goal state config related to the Remote Access.
+    """
+    #
+    # <RemoteAccess>
+    #   <Version/>
+    #   <Incarnation/>
+    #    <Users>
+    #       <User>
+    #         <Name/>
+    #         <Password/>
+    #         <Expiration/>
+    #         <Groups>
+    #           <GroupName/>
+    #           ...
+    #         </Groups>
+    #       </User>
+    #     </Users>
+    #   </RemoteAccess>
+    #
+
+    def __init(self):
+        self.reinitialize()
+
+    def reinitialize(self):
+        """
+        Reset members.
+        """
+        self.Incarnation = None
+        self.Users = []
+        self.xmlText = None
+    
+    def Parse(self, xmlText):
+        """
+        Parse Remote Access configuration.
+        """
+        LogIfVerbose(xmlText)
+        self.reinitialize()
+        self.xmlText = xmlText
+        dom = xml.dom.minidom.parseString(xmlText)
+        node = dom.childNodes[0]
+        if node.localName != "RemoteAccess":
+            Error("RemoteAccess.Parse: root not RemoteAccess")
+
+        incarnationNodes = dom.childNodes.getElementsByTagName("Incarnation")
+        if incarnationNodes is None or len(incarnationNodes) == 0:
+            Error("RemoteAccess.Parse: Incarnation not specified")
+        else:
+            self.Incarnation = incarnationNodes[0].data
+
+        userNodes = dom.childNodes.getElementsByTagName("User")
+        if userNodes is None or len(userNodes) == 0:
+            return self
+        for userNode in userNodes:
+            userAccount = UserAccount()
+            userAccount.Name = userNode.childNodes.getElementsByTagName("Name")[0].data
+            userAccount.EncryptedPassword = userNode.childNodes.getElementsByTagName("Password")[0].data
+            userAccount.Expiration = userNode.childNodes.getElementsByTagName("Expiration")[0].data
+            groupNodes = userNode.childNodes.getElementsByTagname("GroupName")
+            if groupNodes is not None and len(groupNodes) != 0:
+                for groupNode in groupNodes:
+                    userAccount.Groups.append(groupNode.data)
+            self.Users.append(userAccount)
+        return self
+
+    #def Process(self):
+        ##TODO: Process RemoteAccess
```

Comment:
What is this?

Reply:
removed, I'm not sure why changes were made to this file